openmrs-compose.yaml

```diff
@@ -0,0 +1,42 @@
+version: '2.1'
+services:
+  mysql:
+    image: mysql:5.7
+    command: "mysqld --character-set-server=utf8 --collation-server=utf8_general_ci"
+    restart: always
+    ports:
+      - "3306:3306"
+    container_name: openmrs-fhir-mysql
+    healthcheck:
+      test: "exit 0"
+    environment:
+      - MYSQL_DATABASE=openmrs
+      - MYSQL_ROOT_PASSWORD=debezium
+      - MYSQL_USER=mysqluser
+      - MYSQL_PASSWORD=mysqlpw
+      - MYSQL_ROOT_HOST=%    # Allow docker containers to connect to mysql
+    volumes:
+      - ./utils/mysql.cnf:/etc/mysql/conf.d/custom.cnf  # mysql config preconfigured to allow binlog/debezium
+      - ./data:/var/lib/mysql
+      - ./utils/dbdump:/docker-entrypoint-initdb.d # will execute any sql dumped here
+
+  openmrs:
```

Comment:
Apparently GitHub swallowed my comment here so I repeat: When I tested, even after fixing the mysql container issue above, still it did not succeed to start openmrs in a `healthy` state. I have not got into the bottom of this but looking at the logs, it seems it is applying wrong `liquibase` changes. I have pasted one example below. It seems to me that the `liquibase` changes is from [here](https://github.com/openmrs/openmrs-core/blob/4b3d44b89db4c1e6536886ae3f8a400110e9bf0d/api/src/main/resources/org/openmrs/liquibase/updates/liquibase-update-to-latest-2.0.x.xml#L441) but if you look at your `utils/dbdump/backup.sql` the `concept_name` table does not have a `precise` column (because that is already renamed in a future `liquibase` change). Did you experience any issue like this?

Here is the sample error:
```
INFO 9/15/20 7:07 PM:liquibase: Successfully released change log lock
WARN - InitializationFilter$InitializationCompletion$1.run(1650) |2020-09-15 19:07:37,045| Error while trying to update to the latest database version
org.openmrs.util.DatabaseUpdateException: There was an error while updating the database to the latest. file: liquibase-update-to-latest.xml. Error: Migration failed for change set liquibase-update-to-latest.xml::20090316-1008
::vanand:
     Reason: liquibase.exception.DatabaseException: Error executing SQL ALTER TABLE `concept_numeric` MODIFY `precise` SMALLINT NOT NULL DEFAULT 0 DEFAULT 0: Unknown column 'precise' in 'concept_numeric':
          Caused By: Error executing SQL ALTER TABLE `concept_numeric` MODIFY `precise` SMALLINT NOT NULL DEFAULT 0 DEFAULT 0: Unknown column 'precise' in 'concept_numeric':
          Caused By: Unknown column 'precise' in 'concept_numeric'
        at org.openmrs.util.DatabaseUpdater.executeChangelog(DatabaseUpdater.java:154)
        at org.openmrs.web.filter.initialization.InitializationFilter$InitializationCompletion$1.run(InitializationFilter.java:1644)
        at java.lang.Thread.run(Thread.java:748)
Caused by: liquibase.exception.MigrationFailedException: Migration failed for change set liquibase-update-to-latest.xml::20090316-1008::vanand:
     Reason: liquibase.exception.DatabaseException: Error executing SQL ALTER TABLE `concept_numeric` MODIFY `precise` SMALLINT NOT NULL DEFAULT 0 DEFAULT 0: Unknown column 'precise' in 'concept_numeric':
          Caused By: Error executing SQL ALTER TABLE `concept_numeric` MODIFY `precise` SMALLINT NOT NULL DEFAULT 0 DEFAULT 0: Unknown column 'precise' in 'concept_numeric':
          Caused By: Unknown column 'precise' in 'concept_numeric'
        at liquibase.changelog.ChangeSet.execute(ChangeSet.java:347)
        at liquibase.changelog.visitor.UpdateVisitor.visit(UpdateVisitor.java:27)
        at org.openmrs.util.DatabaseUpdater$1OpenmrsUpdateVisitor.visit(DatabaseUpdater.java:191)
        at liquibase.changelog.ChangeLogIterator.run(ChangeLogIterator.java:58)
        at org.openmrs.util.DatabaseUpdater.executeChangelog(DatabaseUpdater.java:220)
        at org.openmrs.util.DatabaseUpdater.executeChangelog(DatabaseUpdater.java:150)
        ... 2 more
Caused by: liquibase.exception.DatabaseException: Error executing SQL ALTER TABLE `concept_numeric` MODIFY `precise` SMALLINT NOT NULL DEFAULT 0 DEFAULT 0: Unknown column 'precise' in 'concept_numeric'
        at liquibase.executor.jvm.JdbcExecutor.execute(JdbcExecutor.java:62)
        at liquibase.executor.jvm.JdbcExecutor.execute(JdbcExecutor.java:104)
        at liquibase.database.AbstractDatabase.execute(AbstractDatabase.java:1091)
        at liquibase.database.AbstractDatabase.executeStatements(AbstractDatabase.java:1075)
        at liquibase.changelog.ChangeSet.execute(ChangeSet.java:317)
        ... 7 more
Caused by: com.mysql.jdbc.exceptions.jdbc4.MySQLSyntaxErrorException: Unknown column 'precise' in 'concept_numeric'
        at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
        at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
        at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
        at com.mysql.jdbc.Util.handleNewInstance(Util.java:425)
        at com.mysql.jdbc.Util.getInstance(Util.java:408)
        at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:944)
        at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3973)
        at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3909)
        at com.mysql.jdbc.MysqlIO.sendCommand(MysqlIO.java:2527)
        at com.mysql.jdbc.MysqlIO.sqlQueryDirect(MysqlIO.java:2680)
        at com.mysql.jdbc.ConnectionImpl.execSQL(ConnectionImpl.java:2480)
        at com.mysql.jdbc.ConnectionImpl.execSQL(ConnectionImpl.java:2438)
        at com.mysql.jdbc.StatementImpl.executeInternal(StatementImpl.java:845)
        at com.mysql.jdbc.StatementImpl.execute(StatementImpl.java:745)
        at liquibase.executor.jvm.JdbcExecutor$1ExecuteStatementCallback.doInStatement(JdbcExecutor.java:92)
        at liquibase.executor.jvm.JdbcExecutor.execute(JdbcExecutor.java:55)
        ... 11 more
```

Reply:
Thanks, @bashir2  for this, working on this!

Reply:
Hi @bashir2,
Thanks for forking this PR and testing out locally! 

I have not been able to reproduce ths issue. However, I think we should do away with `utils/dbdump/backup.sql` and only use the default demo data, to avoid these weird errors . Also we can avoid the /data dir.  

The main purpose of the openmrs-docker-compose is basically to make it easier for a  dev to quickly spin up openmrs with required modules (FHIR + ATOM) + sizeable data to work with. As such can we do away with` utils/dbdump/backup.sql` until we have the deidentified data ready for testing. Also I can also provide the script I used to generate a sizable testdata --> and maybe we should have a separate ticket/issue to create one that is much representative of real data --> the script I used only expounds obs and encounter. 

What do you think?

Reply:
The problem is that even without `backup.sql` I still cannot get into a healthy state for the `openmrs` container. Here is what I did:
1) Remove `utils/dbdump/backup.sql`
2) `docker-compose -f openmrs-compose.yaml up`

The MySQL container runs fine and I can connect to it but there are no tables except  `liquibasechange*` ones. OpenMRS cannot start as its initialization gives many of these errors (below). So now it seems the original tables are not created but the liquibase changes are trying to modify them:
```
openmrs    | INFO 9/22/20 4:08 PM:liquibase: Reading from `liquibasechangelog`                                                                                                                                                    
openmrs    | SEVERE 9/22/20 4:08 PM:liquibase: Change Set liquibase-update-to-latest.xml::1::upul failed.  Error: Error executing SQL ALTER TABLE `person_attribute_type` ADD `edit_privilege` VARCHAR(255): Table 'openmrs.person
_attribute_type' doesn't exist                                                                                                                                                                                                    
openmrs    | liquibase.exception.DatabaseException: Error executing SQL ALTER TABLE `person_attribute_type` ADD `edit_privilege` VARCHAR(255): Table 'openmrs.person_attribute_type' doesn't exist                                
openmrs    |    at liquibase.executor.jvm.JdbcExecutor.execute(JdbcExecutor.java:62)                                                                                                                                              
openmrs    |    at liquibase.executor.jvm.JdbcExecutor.execute(JdbcExecutor.java:104)                                                                                                                                             
openmrs    |    at liquibase.database.AbstractDatabase.execute(AbstractDatabase.java:1091)                                                                                                                                        
openmrs    |    at liquibase.database.AbstractDatabase.executeStatements(AbstractDatabase.java:1075)                                                                                                                              
openmrs    |    at liquibase.changelog.ChangeSet.execute(ChangeSet.java:317)                                                                                                                                                      
openmrs    |    at liquibase.changelog.visitor.UpdateVisitor.visit(UpdateVisitor.java:27)                                                                                                                                         
openmrs    |    at org.openmrs.util.DatabaseUpdater$1OpenmrsUpdateVisitor.visit(DatabaseUpdater.java:191)                                                                                                                         
openmrs    |    at liquibase.changelog.ChangeLogIterator.run(ChangeLogIterator.java:58)                                                                                                                                           
openmrs    |    at org.openmrs.util.DatabaseUpdater.executeChangelog(DatabaseUpdater.java:220)                                                                                                                                    
openmrs    |    at org.openmrs.util.DatabaseUpdater.executeChangelog(DatabaseUpdater.java:150)                                                                                                                                    
openmrs    |    at org.openmrs.web.filter.initialization.InitializationFilter$InitializationCompletion$1.run(InitializationFilter.java:1644)                                                                                      
openmrs    |    at java.lang.Thread.run(Thread.java:748)                                                                                                                                                                           
```

Reply:
Hey @bashir2, as disussed over slack, I have,

1. Removed `utils/dbdump`
2. Removed ./data
3. I have edited openmrs-compose.yaml appropriately to reflect the  above changes.

To test that this is working, clone this branch afresh, and run:

1. docker-compose -f openmrs-compose.yaml down -v 
2. Delete ./data
3. docker-compose -f openmrs-compose.yaml up

should working fine now :)